pkg/standalone-utils/test/GaugeActions.test.ts

```diff
@@ -0,0 +1,659 @@
+    relayerLibrary = await deploy('MockBatchRelayerLibrary', {
+      args: [vault.address, ZERO_ADDRESS, balMinter.address],
+    });
+    relayer = await deployedAt('BalancerRelayer', await relayerLibrary.getEntrypoint());
+
+    // Authorize Relayer for all actions
+    const relayerActionIds = await Promise.all(
+      ['setRelayerApproval', 'manageUserBalance'].map((action) => actionId(vault.instance, action))
+    );
+    await vault.grantPermissionsGlobally(relayerActionIds, relayer);
+
+    // Approve relayer by BPT holder
+    await vault.setRelayerApproval(userSender, relayer, true);
+  });
+
+  // We won't be needing the adder; we just need a staking liquidity gauge where tokens can be deposited / withdrawn.
+  sharedBeforeEach('set up gauge', async () => {
+    const veBalDelegation = await deploy('v2-liquidity-mining/MockVeDelegation');
+
+    const veBalDelegationProxy = await deploy('v2-liquidity-mining/VotingEscrowDelegationProxy', {
+      args: [vault.address, veBAL.address, veBalDelegation.address],
+    });
+
+    const gaugeImplementation = await deploy('v2-liquidity-mining/LiquidityGaugeV5', {
+      args: [balMinter.address, veBalDelegationProxy.address, adaptor.address],
+    });
+    gaugeFactory = await deploy('v2-liquidity-mining/LiquidityGaugeFactory', { args: [gaugeImplementation.address] });
+
+    gauge = await deployedAt(
+      'v2-liquidity-mining/LiquidityGaugeV5',
+      await deployGauge(gaugeFactory, lpToken.address) // No weight cap.
+    );
+
+    // Type weight is ignored in the mock controller.
+    await gaugeController.add_type('Ethereum', 0);
+    await gaugeController.add_gauge(gauge.address, 0); // Type: Ethereum in mock controller.
+  });
+
+  describe('gaugeDeposit', () => {
+    let tokenSender: Account, tokenRecipient: Account;
+
+    context('when using relayer library directly', () => {
+      it('reverts', async () => {
+        expect(
+          relayerLibrary.connect(userSender).gaugeDeposit(gauge.address, userSender.address, gauge.address, fp(1))
+        ).to.be.revertedWith('Incorrect sender');
+      });
+    });
+
+    context('when caller != sender and sender != relayer', () => {
+      it('reverts', async () => {
+        expect(
+          relayerLibrary.connect(userSender).gaugeDeposit(gauge.address, other.address, gauge.address, fp(1))
+        ).to.be.revertedWith('Incorrect sender');
+      });
+    });
+
+    context('when sender does not have enough BPT', () => {
+      it('reverts', async () => {
+        expect(
+          relayer
+            .connect(userSender)
+            .multicall([
+              encodeDeposit({ gauge: gauge, sender: userSender, recipient: userSender, amount: totalLpTokens.add(1) }),
+            ])
+        ).to.be.reverted;
+      });
+    });
+
+    context('when sender has enough BPT', () => {
+      context('sender = senderUser, recipient = senderUser', () => {
+        sharedBeforeEach('', async () => {
```

Comment:
```suggestion
        sharedBeforeEach(async () => {
```

This helper is smart enough to deal with no description being provided :)